lib/key.js

```diff
@@ -80,6 +80,12 @@ function Key(opts) {
 
 Key.formats = formats;
 
+Key.prototype.addPart = function (name, options) {
```

Comment:
Not a fan of this. Please just put together the parts in an object and pass it to the constructor so it can validate everything as needed in there. I don't want a whole new style of constructing keys just for this one format.